pallets/creator-staking/src/types/mod.rs

```diff
@@ -0,0 +1,202 @@
+
+/// Represents an balance amount undergoing the unbonding process.
+/// Since unbonding takes time, it's important to keep track of when and how much was unbonded.
+#[derive(
+Clone, Copy, PartialEq, Encode, Decode, Default, RuntimeDebug, TypeInfo, MaxEncodedLen,
+)]
+pub struct UnlockingChunk<Balance: MaxEncodedLen> {
+    /// Amount being unlocked
+    #[codec(compact)]
+    pub(super) amount: Balance,
+    /// Era in which the amount will become unlocked and can be withdrawn.
+    #[codec(compact)]
+    pub(super) unlock_era: EraIndex,
+}
+
+/// Contains unlocking chunks.
+/// This is a convenience struct that provides various utility methods to help with unbonding
+/// handling.
+#[derive(Clone, PartialEq, Encode, Decode, Default, RuntimeDebug, TypeInfo)]
+pub struct UnbondingInfo<Balance: AtLeast32BitUnsigned + Default + Copy + MaxEncodedLen> {
+    // Vector of unlocking chunks. Sorted in ascending order in respect to unlock_era.
+    unlocking_chunks: Vec<UnlockingChunk<Balance>>,
+}
+
+/// A record of rewards allocated for stakers and creators
+#[derive(PartialEq, Eq, Clone, Default, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct RewardInfo<Balance: HasCompact + MaxEncodedLen> {
+    /// Total amount of rewards for stakers in an era
+    #[codec(compact)]
+    pub stakers: Balance,
+    /// Total amount of rewards for creators in an era
+    #[codec(compact)]
+    pub creators: Balance,
+}
+
+/// A record for total rewards and total amount staked for an era
+#[derive(PartialEq, Eq, Clone, Default, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct EraInfo<Balance: HasCompact + MaxEncodedLen> {
+    /// Total amount of earned rewards for an era
+    pub rewards: RewardInfo<Balance>,
+    /// Total staked amount in an era
+    #[codec(compact)]
+    pub staked: Balance,
+    /// Total locked amount in an era
+    #[codec(compact)]
+    pub locked: Balance,
+}
+
+/// Mode of era-forcing.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub enum Forcing {
+    /// Not forcing anything - just let whatever happen.
+    #[default]
+    NotForcing,
+    /// Force a new era, then reset to `NotForcing` as soon as it is done.
+    /// Note that this will force to trigger an election until a new era is triggered, if the
+    /// election failed, the next session end will trigger a new election again, until success.
+    ForceNew,
+}
+
+/// A list of configuration parameters used to calculate reward distribution portions.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct RewardDistributionConfig {
+    /// Base percentage of reward that goes to stakers
+    #[codec(compact)]
+    pub stakers_percent: Perbill,
+    /// Percentage of rewards that goes to dApps
```

Comment:
```suggestion
    /// Percentage of rewards that goes to creators
```

Reply:
@F3Joule looks like you skipped the suggestion